### PR TITLE
VRM docstrings

### PR DESCRIPTION
**Pull request by dccowan**
# Documentation and Examples for VRM code

Here, we address remaining work outlined in PR #645 ; i.e. documentation and useful examples.

## Details

* from PR #684 
* commits from @dccowan
* reviewed by @thast and @lheagy 

## Highlights

* [x] forward modeling and inversion example with figures
* [x] add properties using properties package
* [x] generate API
* [x] add square loop receiver class

### CODE REVIEW

**Comment by lheagy** [comment]
@thast: would you be willing to review this?

File: examples/12-vrm/plot_vrm_inv.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    reg = Regularization.Simple(mesh=mesh, indActive=actCells, alpha_s=0.25,  cell_weights=W)
+    opt = Optimization.ProjectedGNCG(maxIter=20, lower=0., upper=1e-2, maxIterLS=20, tolCG=1e-4)
+    invProb = InvProblem.BaseInvProblem(dmis, reg, opt)
+    directives = [
+        Directives.BetaSchedule(coolingFactor=2, coolingRate=1),
+        Directives.TargetMisfit()
+    ]
+    inv = Inversion.BaseInversion(invProb, directiveList=directives)
+
+    # m0 = 1e-6*np.ones(actCells.sum())
```
Comment (thast) [inline examples/12-vrm/plot_vrm_inv.py]: Unnecessary line of code

File: SimPEG/VRM/ProblemVRM.py [hunk 2]
```diff
@@ -155,170 +139,387 @@ def _getGeometryMatrix(self, xyzc, xyzh, pp):
...
+                        v1[np.abs(v1) < tol] = np.min(xyzh[:, 1])/tol2
+                        v2 = locs[rr, 1] - by
+                        v2[np.abs(v2) < tol] = -np.min(xyzh[:, 1])/tol2
+                        w1 = locs[rr, 2] - az
+                        w1[np.abs(w1) < tol] = np.min(xyzh[:, 2])/tol2
+                        w2 = locs[rr, 2] - bz
+                        w2[np.abs(w2) < tol] = -np.min(xyzh[:, 2])/tol2
+
+                        Gxx = (
+                            np.arctan((v1*w1)/(u1*np.sqrt(u1**2+v1**2+w1**2)+tol)) -
```
Comment (thast) [inline SimPEG/VRM/ProblemVRM.py]: lots of those lines are too long for the liking of my linter. Do you use one?

File: examples/12-vrm/plot_vrm_inv.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        Ax34 = Fig.add_axes([0.88, 0.05, 0.01, 0.24])
+
+        FS = 12
+        N = x.shape[0]
+
+        invMap = Maps.InjectActiveCells(mesh, actCells, 0.)  # Maps to mesh
+        topoMap = Maps.InjectActiveCells(mesh, topoCells, 0.)
+
+        MAX = np.max(np.r_[xi_true, xi_rec])
+        Cplot11 = mesh.plotSlice(topoMap*xi_true, ind=int((ncz+2*npad)/2-1), ax=Ax11, grid=True, pcolorOpts={'cmap': 'gist_heat_r'})
```
Comment (thast) [inline examples/12-vrm/plot_vrm_inv.py]: lines are too long for pep8 standard

File: tests/em/vrm/vrmfwd_tests.py [hunk 4]
```diff
@@ -265,21 +258,20 @@ def refinefcn(cell):
...
         Survey1 = VRM.Survey(txList)
         Survey2 = VRM.Survey(txList)
         Survey3 = VRM.Survey(txList)
         Survey4 = VRM.Survey(txList)
-        Problem1 = VRM.Problem_Linear(meshObj_Tensor, refFact=2, refRadius=[1.9, 3.6])
-        Problem2 = VRM.Problem_LogUniform(meshObj_Tensor, refFact=2, refRadius=[1.9, 3.6], chi0=mod_chi0_a, dchi=mod_dchi_a, tau1=mod_tau1_a, tau2=mod_tau2_a)
-        Problem3 = VRM.Problem_Linear(meshObj_OcTree, refFact=0)
-        Problem4 = VRM.Problem_LogUniform(meshObj_OcTree, refFact=0, chi0=mod_chi0_b, dchi=mod_dchi_b, tau1=mod_tau1_b, tau2=mod_tau2_b)
+        Problem1 = VRM.Problem_Linear(meshObj_Tensor, ref_factor=2, ref_radius=[1.9, 3.6])
+        Problem2 = VRM.Problem_LogUniform(meshObj_Tensor, ref_factor=2, ref_radius=[1.9, 3.6], chi0=mod_chi0_a, dchi=mod_dchi_a, tau1=mod_tau1_a, tau2=mod_tau2_a)
```
Comment (thast) [inline tests/em/vrm/vrmfwd_tests.py]: it needs some pep8 love

File: examples/12-vrm/plot_vrm_inv.py (hunk 1 above)
Comment (dccowan) [reply examples/12-vrm/plot_vrm_inv.py]: Fixed

File: SimPEG/VRM/ProblemVRM.py (hunk 2 above)
Comment (dccowan) [reply SimPEG/VRM/ProblemVRM.py]: Looked through these. Made some spacing adjustments

File: examples/12-vrm/plot_vrm_inv.py (hunk 3 above)
Comment (dccowan) [reply examples/12-vrm/plot_vrm_inv.py]: Lines no longer cut of in examples

**Comment by thast** [comment]
@dccowan : the examples numbering (12-vrm; 13-Utils) will be in conflict with master as there is now a 13-ip example folder

**Comment by lheagy** [comment]
@dccowan: would you mind updating 
- [ ] the folder naming so that it is `13_ip`, `14_vrm`, `15_utils`
- [ ] and taking a look at the codacy comments, there are a number of low-hanging items that are worth addressing at this stage: https://app.codacy.com/app/thast/simpeg/pullRequest?prid=1503428&/login

File: SimPEG/Directives.py [hunk 5]
```diff
@@ -732,10 +736,10 @@ def endIter(self):
 
                 if reg.eps_p > self.floorEps_p and self.coolEps_p:
                     reg.eps_p /= self.coolEpsFact
-                    print('Eps_p: ' + str(reg.eps_p))
+
                 if reg.eps_q > self.floorEps_q and self.coolEps_q:
                     reg.eps_q /= self.coolEpsFact
-                    print('Eps_q: ' + str(reg.eps_q))
+                    print(reg.eps_q)
```
Comment (lheagy) [inline SimPEG/Directives.py]: @fourndo: can you take a quick look here: do you want to also print out the name of the variable that is being displayed?

File: SimPEG/Directives.py [hunk 6]
```diff
@@ -900,35 +904,39 @@ class UpdatePreconditioner(InversionDirective):
...
+        reg_diag = np.zeros_like(self.invProb.model)
+        m = self.invProb.model
+        for reg in self.reg.objfcts:
+            reg_diag += self.invProb.beta*(
+                (reg.mapping.deriv(m).T*reg.W.T) *
+                (reg.W*reg.mapping.deriv(m))
+            ).diagonal()
 
-            Hdiag = self.opt.JtJdiag + reg_diag
+        Hdiag = self.opt.JtJdiag + reg_diag
```
Comment (lheagy) [inline SimPEG/Directives.py]: @fourndo: here too

**Comment by lheagy** [comment]
I am going to close this in favour of #708, they contain the same changes, and we can just do one pr review there

**Pull request closed without merging**